### PR TITLE
Write to TensorBoard every x samples.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
### Summary

Closes #7617 .
This PR is an update of #7617. Credit goes to @sameermanek for his work.

The goal of the PR is to be able to see the metrics and losses in a per-batch basis instead of a per-epoch basis.

### Related Issues

#6692

### PR Overview

**This PR is backward compatible** 

A new argument has been added to the constructor of the TensorBoard callback. Here is the description that I added in the docs:

>write_step: 
`'batch'` or `'epoch'` or integer. When using `'batch'`, writes
            the losses and metrics to TensorBoard after each batch. The same
            applies for `'epoch'`. If using an integer, let's say `10000`,
            the callback will write the metrics and losses to TensorBoard every
            10000 samples. Note that writing too frequently to TensorBoard
            can slow down your training.

Feedback welcome. Especially from @fchollet  since the API has been changed.

- [x] This PR requires new unit tests [y/n] (make sure tests are included)
- [x] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [x] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: keras/callbacks.py [hunk 1]
```diff
@@ -715,6 +715,12 @@ class TensorBoard(Callback):
             `embeddings_layer_names`. Numpy array (if the model has a single
             input) or list of Numpy arrays (if the model has multiple inputs).
             Learn [more about embeddings](https://www.tensorflow.org/programmers_guide/embedding)
+        write_step: `'batch'` or `'epoch'` or integer. When using `'batch'`, writes
```
Comment (fchollet) [inline keras/callbacks.py]: There's already a `embeddings_freq` argument, and this new argument `write_step` is not consistent with it. Maybe `write_freq` or `update_freq` would be better. Thoughts?

Comment (gabrieldemarmiesse) [reply keras/callbacks.py]: Naming things is as hard as ever.
I like `update_freq` since we "abstract" the writing to the file in this case. The user doesn't really care (take this with a grain of salt) what happens to the file. He/she wants to see the plot being updated. That's what matters.

**Comment by ahundt** [comment]
LGTM, a useful feature and a nice improvement over the pr it is based on. The new param names + options are much better.

**Comment by gabrieldemarmiesse** [comment]
Thanks. I don't think I'll make the PR in the Tensorflow repo. I'll let someone else do it if you don't mind.

**Comment by bersbersbers** [comment]
I just recently switches from Keras to tf.keras, and noticed that TensorBoard summaries looked a little bit different there: With tensorflow-gpu==1.11.0 and tensorboard==1.11.0, I no longer have data in val_loss, but in epoch_val_loss; and in addition to loss, there are epoch_loss and batch_loss. Is that a difference between the two Keras version? Will it impact porting of this feature here to tf.keras?
